lua/hydra/defaults.lua

```diff
@@ -0,0 +1,24 @@
+local M = {}
+
+---@type hydra.Config
+M.default_config = {
+  debug = false,
+  exit = false,
+  foreign_keys = nil,
+  color = "red",
+  timeout = false,
+  invoke_on_body = false,
+  hint = {
+    show_name = true,
+    position = { "bottom" },
+    offset = 0,
+  },
+}
+
+--- Setup the default configuration
+--- @param opts hydra.OptionalConfig
+M.setup = function(opts)
```

Comment:
The setup function would be called by the user.

This allows setting common values in one place. For example if someone wants a custom border on all their hydras, they can call this method once instead of passing the border each time they make a new Hydra.

Reply:
I misunderstood what the code was doing. The setup function is unused in the codebase and that had me quite confused (hence the request for clarification).

I don't know that I like burying the setup inside "defaults". Given that hydra uses that wonky "class" system for the `Hydra` object, is there a way we can expose `setup` from hydra itself? IE, instead of
```lua
require("hydra.defaults").setup()
```

We have 
```lua
require("hydra").setup()
```

Its not a huge deal, but I do feel that if we are going to do this, we should have setup be in the "standard" location. If not, we need to update the documentation explaining how to call it, and likely updating the lazy installation to show how to call setup since `config = true` wont work there (without using whatever the flag is to indicate that setup is located elsewhere).

Additionally, we can run setup as many times as we want and each time will override the global configuration. Should we lock it so that it can only be ran once? Set a global `vim.g.__hydra_setup` value and check for that before running the configuration merge or something?

Reply:
> a way we can expose `setup` from hydra itself

I kinda assumed this wouldn't work, but I just tried it out and it does. happy to make that change.

> Additionally, we can run setup as many times as we want and each time will override the global configuration

I don't think this is a big deal, and would actually rather it behave this way.

Reply:
> I kinda assumed this wouldn't work, but I just tried it out and it does. happy to make that change.

Awesome!

> I don't think this is a big deal, and would actually rather it behave this way.

If you're good with it, I am good with it :) I just wanted to bring it up in case that wasn't the "desired" behavior